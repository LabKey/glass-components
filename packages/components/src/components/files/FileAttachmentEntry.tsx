
import * as React from 'react';

interface Props {
    allowDelete?: boolean
    onDelete?: (name: string) => any
    name: string
    key?: string
    titleText?: string
}

export class FileAttachmentEntry extends React.PureComponent<Props, any> {

    static defaultProps = {
        allowDelete: true,
        titleText: "Remove file"
    };

    render() {
        const { allowDelete, onDelete, key, name, titleText } = this.props;

        return (
            <div key={key || name} className="attached-file--container">
                {allowDelete && <span
                    className="fa fa-times-circle file-upload__remove--icon"
                    onClick={() => onDelete(name)}
                    title={titleText}
                />}
                <span className="fa fa-file-text attached-file--icon attached-file--bottom-spacing"/>
                {name}
            </div>
        )
    }
}
