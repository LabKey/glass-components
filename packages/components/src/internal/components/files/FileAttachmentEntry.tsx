import React, { PureComponent } from 'react';

interface Props {
    allowDelete?: boolean;
    onDelete?: (name: string) => void;
    name: string;
    downloadUrl?: string;
    deleteTitleText?: string;
}

export class FileAttachmentEntry extends PureComponent<Props, any> {
    static defaultProps = {
        allowDelete: true,
        deleteTitleText: 'Remove file',
    };

    render() {
        const { allowDelete, downloadUrl, onDelete, name, deleteTitleText } = this.props;
        return (
            <div key={name} className="attached-file--container">
                {allowDelete && (
                    <span
                        className="fa fa-times-circle file-upload__remove--icon"
                        onClick={() => onDelete(name)}
                        title={deleteTitleText}
                    />
                )}
                <span className="fa fa-file-text attached-file--icon attached-file--bottom-spacing" />
                {downloadUrl ? (
                    <strong>
                        <a href={downloadUrl} title={name}>
                            <div className="file-listing-filename">{name}</div>
                        </a>
                    </strong>
                ) : (
                    name
                )}
            </div>
        );
    }
}
