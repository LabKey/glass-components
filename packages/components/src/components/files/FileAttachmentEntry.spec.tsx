import React from 'react';
import { FileAttachmentEntry } from './FileAttachmentEntry';
import { shallow } from 'enzyme';

describe("<FileAttachmentEntry>", () => {
    test("default props", () => {
        const wrapper = shallow(<FileAttachmentEntry
            onDelete={jest.fn()}
            name={"Test files"}
        />);
        expect(wrapper.find("span.fa-times-circle")).toHaveLength(1);
        expect(wrapper.text()).toBe("Test files");
        expect(wrapper).toMatchSnapshot();
    });
    test("no deletion", () => {
        const wrapper = shallow(<FileAttachmentEntry
            allowDelete={false}
            name={"Test files"}
        />);
        expect(wrapper.find("span.fa-times-circle")).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });
    test("with deleteTitleText", () => {
        const wrapper = shallow(<FileAttachmentEntry
            onDelete={jest.fn()}
            deleteTitleText={"Delete me"}
            name={"Test files"}
        />);
        expect(wrapper.find("span.fa-times-circle")).toHaveLength(1);
        expect(wrapper.find({title: "Delete me"})).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
    test("with downloadUrl", () => {
        const wrapper = shallow(<FileAttachmentEntry
            allowDelete={false}
            downloadUrl={"http://get/me/my/file"}
            name={"Test files"}
        />);
        expect(wrapper.find('a[href="http://get/me/my/file"]')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
});
