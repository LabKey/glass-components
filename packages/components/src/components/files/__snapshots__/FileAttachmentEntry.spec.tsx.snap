// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileAttachmentEntry> default props 1`] = `
<div
  className="attached-file--container"
  key="Test files"
>
  <span
    className="fa fa-times-circle file-upload__remove--icon"
    onClick={[Function]}
    title="Remove file"
  />
  <span
    className="fa fa-file-text attached-file--icon attached-file--bottom-spacing"
  />
  Test files
</div>
`;

exports[`<FileAttachmentEntry> no deletion 1`] = `
<div
  className="attached-file--container"
  key="Test files"
>
  <span
    className="fa fa-file-text attached-file--icon attached-file--bottom-spacing"
  />
  Test files
</div>
`;

exports[`<FileAttachmentEntry> with deleteTitleText 1`] = `
<div
  className="attached-file--container"
  key="Test files"
>
  <span
    className="fa fa-times-circle file-upload__remove--icon"
    onClick={[Function]}
    title="Delete me"
  />
  <span
    className="fa fa-file-text attached-file--icon attached-file--bottom-spacing"
  />
  Test files
</div>
`;

exports[`<FileAttachmentEntry> with downloadUrl 1`] = `
<div
  className="attached-file--container"
  key="Test files"
>
  <span
    className="fa fa-file-text attached-file--icon attached-file--bottom-spacing"
  />
  <strong>
    <a
      href="http://get/me/my/file"
      title="Test files"
    >
      <div
        className="file-listing-filename"
      >
        Test files
      </div>
    </a>
  </strong>
</div>
`;
